SonarRunner.Shim/PropertiesFileGenerator.cs

```diff
@@ -82,6 +89,35 @@ public static ProjectInfoAnalysisResult GenerateFile(AnalysisConfig config, ILog
             return result;
         }
 
+        private static void FixSarifReport(ILogger logger, IEnumerable<ProjectInfo> projects, IRoslynV1SarifFixer fixer)
+        {
+
+            // attempt to fix invalid project-level SARIF emitted by Roslyn 1.0 (VS 2015 RTM)
+            foreach (ProjectInfo project in projects)
+            {
+                Property propertyToRemove = null;
+                foreach (Property projectProperty in project.AnalysisSettings)
```

Comment:
That's the point: if you use project.AnalysisSettings.TryGetValue(key, out value), you won't be in the middle of a for-each.